test-workloads/src/bin/tellus.rs

```diff
@@ -637,6 +585,8 @@ extern "C" fn kernel_init(hart_id: u64, fdt_addr: u64) {
                             zero_pages_added += 1;
                         }
                         GUEST_SHARED_PAGES_START_ADDRESS..=GUEST_SHARED_PAGES_END_ADDRESS => {
```

Comment:
As discussed the TSM will pass the memory region information from run ECALL exit and the host can cache this information. I am thinking the host should use this information to distinguish between shared page fault and zero page fault ?

Reply:
This is just a test program where it's easier to hard-code everything :) But yes, I've updated this to intercept the address of the regions added by the guest and use them for address-matching.

Reply:
I was not suggesting to modify tellus :). I was just clarifying the usage from host implementation perspective.
I see you have already modified the tellus code. Sorry for the extra work ;)